### ISSUE TITLE
Satisforestry 29d Registration Error after fresh Install

### ISSUE THREAD

**Issue by TheNodop** [ReikaKalseki/Reika_Mods_Issues#2786]
Tried to test new Satisforestry version on a fresh and clean install of Minecraft with Forge 1614. (Only DragonAPI & Satisforestry)

Now Satisforestry throws an registration error. Maybe this is the same as the type error, just rephrased.
But I'm not sure so here is a new ticket.

The log this time is attached and as pastebin, I do not know which one you prefer.

https://pastebin.com/sz1kbNDD
[latest.log](https://github.com/ReikaKalseki/Reika_Mods_Issues/files/6215290/latest.log)

Edit: If I can help with anything, like hardware specs, java version or something else feel free to ask.

**Comment by ReikaKalseki** [comment]
Reika.DragonAPI.Exception.RegistrationException: SATISFORESTRY has a registration error:
Configs could not be loaded! Delete them and try again.
Contact Reika immediately!

**Comment by TheNodop** [comment]
I have read that.. This happens all the time.
I delete the configs multiple Times, the problem stays the same.

**Comment by ReikaKalseki** [comment]
Show me a config that is failing to load.

**Comment by TheNodop** [comment]
Thank you for coming back to this so fast.
This one is freshly generated (deleted the complete "Reika" Folder), still only DragonAPI and Satisforestry installed.

This is the content of "Satisforestry.cfg", it's so short I just paste it here:
# Configuration file

"control setup" {
    I:"Cave Mob Spawn Multiplier"=1
    B:"Enable Simple Automation for Resource Node"=false
    I:"Pink Forest Biome ID"=144
}

And there are 4 lua files inside the satisforestry_Files Folder, base,doggo,ores and resources.
For future reference: Is pastebin or an attachement prefered?

**Comment by MalTeeez** [comment]
> For future reference: Is pastebin or an attachement prefered?

I think Pastebin or Gist are best (

> https://github.com/ReikaKalseki/Reika_Mods_Issues/blob/master/CONTRIBUTING.md

)

**Comment by AssasinAsha** [comment]
I also get this bug. Some testing results:

Clearing the folder from all files causes a message "No resource config file found!" with an admonishment not to post this as error.

Adding "resource.lua" file and only that file causes NPEs at "Reika.Satisforestry.Config.BiomeConfig.loadFiles" until both the "ores.lua" and "doggo.lua" files are put in (Affected lines are 250 and 255, depending on which file is missing).

When the above three files are in the folder, I get the crash from the issue.

Removing "resource.lua" at this point causes the first message again.

The presence of "base.lua" file doesn't seem to matter.

Obvious workaround is to delete the folder before starting the client and before each world load. But if I do that I get this crash when nearing the biome:
```
java.lang.NullPointerException: Ticking memory connection
	at Reika.Satisforestry.Blocks.BlockResourceNode$TileResourceNode.generate(BlockResourceNode.java:134)
	at Reika.Satisforestry.Biome.Biomewide.UraniumCave$ResourceNodeRoom.generateResourceNodeAt(UraniumCave.java:1091)
	at Reika.Satisforestry.Biome.Biomewide.UraniumCave$ResourceNodeRoom.generateResourceNode(UraniumCave.java:1085)
	at Reika.Satisforestry.Biome.Biomewide.UraniumCave.generate(UraniumCave.java:211)
	at Reika.Satisforestry.Biome.Biomewide.BiomewideFeatureGenerator.generateUniqueCenterFeatures(BiomewideFeatureGenerator.java:65)
	at Reika.Satisforestry.Biome.DecoratorPinkForest.func_150513_a(DecoratorPinkForest.java:120)
	at Reika.DragonAPI.Instantiable.Worldgen.StackableBiomeDecorator.func_150512_a(StackableBiomeDecorator.java:29)
	at net.minecraft.world.biome.BiomeGenBase.func_76728_a(BiomeGenBase.java:339)
	at Reika.DragonAPI.Auxiliary.WorldGenInterceptionRegistry.runBiomeDecorator(WorldGenInterceptionRegistry.java:229)
	at net.minecraft.world.gen.ChunkProviderGenerate.func_73153_a(ChunkProviderGenerate.java:421)
	at net.minecraft.world.gen.ChunkProviderServer.func_73153_a(ChunkProviderServer.java:279)
	at net.minecraft.world.chunk.Chunk.func_76624_a(Chunk.java:1045)
	at net.minecraft.world.gen.ChunkProviderServer.originalLoadChunk(ChunkProviderServer.java:190)
	at net.minecraft.world.gen.ChunkProviderServer.loadChunk(ChunkProviderServer.java:131)
	at net.minecraft.server.management.PlayerManager$PlayerInstance.<init>(PlayerManager.java:370)
	at net.minecraft.server.management.PlayerManager.func_72690_a(PlayerManager.java:97)
	at net.minecraft.server.management.PlayerManager.func_72691_b(PlayerManager.java:190)
	at net.minecraft.server.management.PlayerManager.func_72685_d(PlayerManager.java:267)
	at net.minecraft.server.management.ServerConfigurationManager.func_72358_d(ServerConfigurationManager.java:313)
	at net.minecraft.network.NetHandlerPlayServer.func_147347_a(NetHandlerPlayServer.java:404)
	at net.minecraft.network.play.client.C03PacketPlayer.func_148833_a(SourceFile:137)
	at net.minecraft.network.play.client.C03PacketPlayer$C04PacketPlayerPosition.func_148833_a(SourceFile:63)
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:212)
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:165)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:659)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:547)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:111)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:427)
	at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685)
```
Here's the line from log just before the above crash, seems relevant:
```
[Server thread/INFO] [FML]: SATISFORESTRY: Generating biome cave @ [-519.5, 51.5, -1056.5], from rivers [[-527, 0, -920], [-353, 0, -966], [-436, 0, -961], [-607, 0, -986], [-562, 0, -963], [-676, 0, -984], [-692, 0, -1031]] and cache null
```

Although I wonder if that whole crash is relevant here. Probably not, but hey! It's something at least.

Hope that helps some! I can try more things if you have suggestions.

**Comment by ReikaKalseki** [comment]
@TheNodop That is not the config I wanted - I wanted the ones that are failing to load, ie those four you mentioned but did not post.

@AssasinAsha You need a valid config to generate the biome - it cannot generate with no data.

**Comment by ReikaKalseki** [comment]
Never mind, #2772 solved it.
